action/protocol/staking/protocol.go

```diff
@@ -387,25 +400,32 @@ func (p *Protocol) Commit(ctx context.Context, sm protocol.StateManager) error {
 
 // Handle handles a staking message
 func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.StateManager) (*action.Receipt, error) {
-	featureWithHeightCtx := protocol.MustGetFeatureWithHeightCtx(ctx)
-	height, err := sm.Height()
+	csm, err := p.constructCandidateStateManager(ctx, sm)
 	if err != nil {
 		return nil, err
 	}
-	csm, err := NewCandidateStateManager(sm, featureWithHeightCtx.ReadStateFromDB(height))
+
+	return p.handle(ctx, act, csm)
+}
+
+func (p *Protocol) constructCandidateStateManager(ctx context.Context, sm protocol.StateManager) (CandidateStateManager, error) {
```

Comment:
why need this change? seems no real change instead of just adding this new func?

Reply:
it's reused in two places